src/battle_setup.c

```diff
@@ -745,6 +746,8 @@ u8 BattleSetup_GetTerrainId(void)
     s16 x, y;
 
     PlayerGetDestCoords(&x, &y);
+    if (I_FISHING_ENVIRONMENT >= GEN_4 && gIsFishingEncounter)
+        GetXYCoordsOneStepInFrontOfPlayer(&x, &y);
```

Comment:
Is there a reason this doesn't just override the previous function? Now there's two functions being called with one's results immediately overwriting the other's

Reply:
Oh, I can change that